tests/cmsis_rtos_v1/src/thread_apis.c

```diff
@@ -106,11 +108,22 @@ void test_thread_prio(void)
 
 	status = osThreadTerminate(id3);
 	zassert_true(status == osOK, "Error terminating thread3");
+
+	/* Try to set unsupported priority and assert failure */
+	status = osThreadSetPriority(osThreadGetId(), osPriorityDeadline);
+	zassert_false(status == osOK, "Invalid priority set");
+
```

Comment:
Better to move the above statements to "thread3()" function as that is where all tests related to priority are happening. Also, it's better to check the return value for the exact return value i.e. in this case, check for osErrorValue. The statement can be something like
zassert_true(status == osErrorValue, "Something's wrong with osThreadSetPriority!");